packages/office-ui-fabric-react/src/components/FocusZone/FocusZone.tsx

```diff
@@ -36,6 +36,7 @@ const LARGE_NEGATIVE_DISTANCE_FROM_CENTER = -999999999;
 const _allInstances: {
   [key: string]: FocusZone;
 } = {};
+const _outerZones: Set<FocusZone> = new Set();
```

Comment:
The usage of `Set` throughout this class works for IE 11 compat judging by https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set#Browser_compatibility

Unsure if we are concerned about future iterations using an unsupported method breaking IE 11 cc: @ecraig12345

Reply:
I am a little nervous on `Set` usage. I can try on IE11 really quick to make sure we are good.

Reply:
I checked and the methods you're currently using are compatible.

Reply:
Yup i checked on ie11 directly, looks good

Reply:
Please do try it out in live IE before checking in. Also, once #8414 to change the libs goes in, you'll have to add a file-level declaration if you want to use Set. Advantage of that is you can limit it to only include the actual supported methods (and have comments warning against adding more methods...).

```ts
// Declare Set at file scope to avoid including the lib es2015.collection in tsconfig,
// so people can't accidentally use the ES6 map/set classes elsewhere and break IE 11.
// (This declaration only includes methods which work in IE 11.)
declare class Set<T> {
  public readonly size: number;
  constructor();
  public add(value: T): void;
  public clear(): void;
  public delete(value: T): boolean;
  public forEach(callbackfn: (value: T, value2: T, set: Set<T>) => void, thisArg?: any): void;
  public has(value: T): boolean;
}
```

Reply:
@ecraig12345 if IE11 already supports this, shouldn't we just have a universal declaration file? It would be bad to have this declare everywhere.

Reply:
(e.g. maybe you could just add an ie11 allow declaration file in your PR that includes this, then we have one place to add these.

Reply:
At least to me it seems simpler to (mostly) just pretend that Set doesn't exist at all yet, and only add declarations where we really need to use it. That way we don't have people confused why they can't use all the methods? Though now that I write it out I'm not sure that's simpler.

Reply:
Hmm. Declaring set at the top of FocusZone.tsx seems like a code smell.

Declaring it centrally seems ok, even if it is a subset; the confusion of why only a subset can be traced to the definition if someone cares about the history, comments in the declares help. Right?

I don't really think we want to pretend it doesn't exist; it does. Plus once IE11 is dead, deleting the declaration polyfill from one place and upgrading the lib would be a much better cleanup than to find individual declarations.

Reply:
Yeah that's probably right. Honestly before this discussion, I hadn't realized that Set existed in any sort of usable from in IE.

Reply:
I appreciate this callout! Thanks for keeping things honest